sched/queue/memory/simple.go

```diff
@@ -0,0 +1,81 @@
+package memory
+
+// In-memory Scheduler Queue
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/sched"
+	"github.com/scootdev/scoot/sched/queue"
+	"time"
+)
+
+func NewSimpleQueue() (queue.Queue, chan sched.Job) {
+	q := &simpleQueue{}
+	q.outCh = make(chan sched.Job, 1)
+	q.inCh = make(chan enqueueReq)
+	go q.loop()
+	return q, q.outCh
+}
+
+type simpleQueue struct {
+	nextID int
+	inCh   chan enqueueReq
+	outCh  chan sched.Job
+}
+
+type enqueueReq struct {
+	job      sched.Job
+	resultCh chan enqueueResult
+}
+
+type enqueueResult struct {
+	jobID string
+	err   error
+}
+
+func (q *simpleQueue) Enqueue(job sched.Job) (string, error) {
```

Comment:
Thought is the validation of a Job something we could provide as a function associated with Queue interface?  Seems like all Queue implementations will have to implement this